design/ceph/multus-network.md

```diff
@@ -90,111 +90,106 @@ interface will be added to the host network namespace for all nodes that will ru
-for the case where only the mover pod is being restarted, but it is very difficult to detect the
-case where the mover is merely being restarted versus when the holder is also being removed without
-possible race conditions between which pod might be stopped first by the Kubernetes API server
-during a drain event. Therefore, focus on the simplest working implementation (described above)
-instead of risking leaving the interface copy in host net namespace which could cause issues.
-
-If an error occurs in the mover during network migration, it will fail and re-try migration until
-the operation succeeds. If necessary, the mover will try to remove a partially-copied interface.
-
-Restarting a node will cause the multus interface in the host namespace to go away. On restart, the
-holder pod will get a new interface, and the mover will copy it into the host networking namespace again.
-
-When a new node is added, holder and mover pods are added to it by their DaemonSets, and the
-move/copy process described above occurs on the node.
-
-The holder and mover DaemonSets should be deleted when the CSI driver components are removed. Both
-the termination of the holder pods as well as the mover pods triggers the mover to remove the multus
-interfaces from the host network namespace of a given node.
-
-The initial implementation of this design will be limited to supporting a single CephCluster with
-Multus until we can be sure that the CSI plugin can support multiple migrated interfaces as well as
-interfaces that are added and removed dynamically. This limitation will be enforced by allowing only
-a single instance of the holder DaemonSet. A possible (future) partial implementation may be
-possible by restarting the CSI plugin Pods when network interfaces are added or removed.
-
-**Known issue:** the Docker container runtime does not use Linux's native `/var/run/netns`
-directory. This mitigation is known to work on cri-o runtime but not Docker. Therefore, this feature
-will be disabled by default and enabled optionally by the
-`ROOK_CSI_MULTUS_USE_HOLDER_MOVER_PATTERN=true` variable in the Rook-Ceph operator's config.
-
-A previous version of the CSI proposal had the holder Pods creating "setup" and "teardown"
-Kubernetes Jobs for migrating/un-migrating the multus networks. This design was rejected since new
-Jobs wouldn't be able to be created if the Kubernetes namespace were in "Terminating" state.
+When the CSI-sidecars DaemonSet is updated, the `cephcsi` binary will have a new version and
+potentially new arguments as well. The new `cephcsi` binary is copied (from the `copy-binary`
+init container) in the Kubelet HostPath volume directory.
+If copying fails, the container will keep retrying until the operation succeeds.
+On the side of the CSI-plugin DaemonSet, the directory will be watched for changes and will
+accordingly kill and start the new binary process. If the new binary process fails to start, the
```

Comment:
It would be good to clarify: under what condition is the process killed? under what condition is it started? 

IIUC, it is killed and immediately started once a new binary appears, correct?